### PR TITLE
Fix some of the failures in routes.spec.js

### PR DESCRIPTION
**Pull request by n-riesco**
@chriddyp @tarzzz Unfortunately, this PR doesn't fix all the failures. This is what I've done:
- ensured that `routes.js` and `authorization.js` always invoke `next()` (especially in those handlers that launch promises).
- ensured servers have been closed after each spec completes (so that it's safe to start the next spec).
- ensured cookies are cleared before running a new spec.
- refactored all the specs so that they don't silence thrown exceptions.
- tightened some of the specs (some of the specs where passing when they should've failed).
- I've disabled the specs for impala in `routes.spec.js` because they fail systematically inside the impala connector (I'll investigate the problems with this connector after this PR is merged).

There are still intermittent failures that I need to debug, but as I can't reproduce them very often, progress is slow.

### CODE REVIEW

File: test/backend/routes.spec.js [hunk 1]
```diff
@@ -205,76 +217,59 @@ describe('Authentication - ', () => {
...
 
-    it('does not allow access to settings when not logged in', function(done) {
-        GET('settings')
-        .then(res => res.json().then(json => {
+    it('backend does not allow access to settings when not logged in', function() {
+        return GET('settings').then(res => res.json().then(json => {
             assert.equal(res.status, 401);
             assert.deepEqual(json, {
-                'error': {'message': 'Please login to access this page.'}
+                error: {message: 'Please login to access this page.'}
```
Comment (tarzzz) [inline test/backend/routes.spec.js]: 👍

File: test/backend/routes.spec.js [hunk 2]
```diff
@@ -294,22 +306,22 @@ describe('Authentication:', () => {
             // This ensures that any requests to plotly fail so that we can catch them
             saveSetting('PLOTLY_API_DOMAIN', 'bad-domain.plot.ly');
 
-            return GET('settings').then(res => res.json().then(json => {
+            return wait(1000).then(() => GET('settings').then(res => res.json().then(json => {
```
Comment (tarzzz) [inline test/backend/routes.spec.js]: Any particular reason for adding this wait?

Comment (n-riesco) [reply test/backend/routes.spec.js]: so that the token expires.

**Comment by tarzzz** [comment]
Looks good to me @n-riesco .. Thanks for making those changes..

**Comment by n-riesco** [comment]
@tarzzz I'll take your comment as a :dancer: and merge.

**Pull request merged**